### PR TITLE
Added callbacks as option and attributes

### PR DESCRIPTION
**Pull request by jsalbert**
I just added some more options
- List of callbacks (e.g. variable learning rates)
- List of metrics
- Workers argument for multiprocessing, max_queue_size
- Possibility of adding custom loss function
- Class Weights

### CODE REVIEW

File: keras_trainer/trainer.py [hunk 1]
```diff
@@ -32,6 +34,11 @@ def __init__(self, model_spec, dictionary, train_dataset_dir, val_dataset_dir, o
         self.val_generator = options.pop('val_generator', None)
         self.top_layers = options.pop('top_layers', None)
         self.optimizer = options.pop('optimizer', None)
+        self.callback_list = options.pop('callback_list', list())
```
Comment (stephensolis) [inline keras_trainer/trainer.py]: `list()` -> `[]`
